cloud-controller-manager/do/metrics.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2020 DigitalOcean
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package do
+
+import "github.com/prometheus/client_golang/prometheus"
+
+type metrics struct {
+	host                 string
+	apiRequestCounter    prometheus.CounterVec
+	apiRequestDuration   prometheus.HistogramVec
+	runLoopsStarted      prometheus.Counter
+	numberOfReconciles   prometheus.Counter
+	eventHandlerDuration prometheus.Histogram
+}
+
+// create metrics
+var (
+	apiRequestCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: "firewall",
+			Name:      "api_request_total",
+			Help:      "This is a counter for tracking firewall API requests.",
+		},
+		[]string{"http_code", "http_method", "success"},
+	)
+	apiRequestDuration = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: "firewall",
+			Name:      "api_request_duration",
```

Comment:
I'm going to change the other one (`api_request_total`) to plural. And make this one `api_request_duration_seconds`. This better aligns with the naming guidelines. And since this is `http`, I will update `api` to be `http` instead.

Thank you for the link @timoreimann !